### ISSUE TITLE
Move interface-specific functions into the interface repos

### ISSUE THREAD

**Issue by imciner2** [osqp/osqp#370]
Currently, there are parts of the main library code that are tailored to each specific interface we build (i.e. Python, Matlab and R). This means the main OSQP CMake file must try searching for those interface libraries, and sometimes means changes to fix interfaces must happen in both the main library and the interface code.

We should move these interface-specific code pieces into the wrapper repositories when possible. The first place we can do this is the memory management functions defined in `global_opts.h` by using the custom memory management function built into the main library.

Future work is still needed to do this for the printing functions though (it needs a similar custom print function interface).

**Comment by gbanjac** [comment]
I agree. We should also disable `EMBEDDED` in the main CMake file as we will generate a tailored one for an embedded device.
